### PR TITLE
[PyTorch] Fix Transformers, Seq2Seq Predictions

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
This PR addresses and fixes the bug mentioned in #1484 .
Since earlier the model was still in `train` mode, the randomness due to dropout kicked in. Now the model is set to `eval` mode for inference in the function `predict_s2s_ch9`. This should fix the transformers as well as seq2seq(which apparently also had the bug but was not clearly visible) sections.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1528/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1528/
